src/tests/index.js

```diff
@@ -0,0 +1,262 @@
+/*global describe, it, beforeEach, before*/
+'use strict'
+const assert = require('assert')
+const Promise = require('bluebird')
+const ServiceManager = require('../lib/service-manager')
+
+const notarySecretKey = 'ZVx0mSsB4+pNdP2q0qq2wEyHKBtoiw2co05Db0sD2Ao6a6skoSHJHAhS5uCKkR8bCu/t3IqKI0jsd3tfbndMcQ=='
+const notaryPublicKey = 'OmurJKEhyRwIUubgipEfGwrv7dyKiiNI7Hd7X253THE='
+
+const services = new ServiceManager(process.cwd())
+
+before(function * () {
+  yield services.startLedger('ledger1', 3001)
+  yield services.startLedger('ledger2', 3002)
+  yield services.startLedger('ledger3', 3003)
+
+  yield services.startConnector('mark', 4001, {
+    pairs: [
+      ['USD@http://localhost:3001', 'USD@http://localhost:3002'],
+      ['USD@http://localhost:3002', 'USD@http://localhost:3001']
+    ],
+    credentials: {
+      'http://localhost:3001': makeCredentials('http://localhost:3001', 'mark'),
+      'http://localhost:3002': makeCredentials('http://localhost:3002', 'mark')
+    }
+  })
+
+  yield services.startConnector('mary', 4002, {
+    pairs: [
+      ['USD@http://localhost:3003', 'USD@http://localhost:3002'],
+      ['USD@http://localhost:3002', 'USD@http://localhost:3003']
+    ],
+    credentials: {
+      'http://localhost:3002': makeCredentials('http://localhost:3002', 'mary'),
+      'http://localhost:3003': makeCredentials('http://localhost:3003', 'mary')
+    }
+  })
+
+  yield services.startNotary('notary1', 6001, {
+    secretKey: notarySecretKey,
+    publicKey: notaryPublicKey
+  })
+})
+
+// Run this before each test
+beforeEach(function * () {
```

Comment:
Thinking about it more, I think it might be best to restart all the components between each test case for now because non-interference between test cases is probably more important than runtime (and we can optimize later with a reset endpoint if runtime becomes an issue). In general I consider test case interference to be a "bug" in the tests, though in this case it is fairly minimal because most of the state is being reset, but still I think its technically a bug.